### PR TITLE
Webpack

### PR DESCRIPTION
**Pull request by lusbenjamin**
Moves our build process from directly invoking `babel` to using `webpack`. This should probably be a breaking change aimed at v3.0.

Resources:
- https://webpack.js.org/configuration/
- https://github.com/Devtography/electron-react-typescript-webpack-boilerplate
- https://www.codementor.io/randyfindley/how-to-build-an-electron-app-using-create-react-app-and-electron-builder-ss1k0sfer

### Done
- Build all of the TS
- move static assets into webpack
- Get all of the imports to work
- Keep `electron-builder` working

### Future Optimizations
- Only ship the contents of `lib` (requires file reorg?)
- hook up a local server and try hot reload hotness

### TODO
- [ ] more manual testing of dev mode
- [ ] test distribution end-to-end
- [ ] fix all dynamic references to static assets, for example the land icon in the overlay is currently broken for the "compact lands" feature due to this: https://github.com/Manuel-777/MTG-Arena-Tool/blob/master/src/shared/CardTile.tsx#L349

### CODE REVIEW

File: src/window_main/renderer-util.js [hunk 1]
```diff
@@ -43,7 +43,7 @@ import {
 } from "../shared/util";
 import ReactDOM from "react-dom";
 import createShareButton from "./createShareButton";
-import { force_open_settings } from "./tabControl";
+import { forceOpenSettings } from "./tabControl";
```
Comment (lusbenjamin) [inline src/window_main/renderer-util.js]: we should port this bugfix

File: webpack.config.js [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+        loader: "babel-loader"
+      },
+      {
+        test: /\.(ts|tsx)$/,
+        include: /src/,
+        loader: "ts-loader"
+      }
+    ]
+  },
+  node: { __dirname: false, __filename: false },
```
Comment (lusbenjamin) [inline webpack.config.js]: this unassuming line may be one of the more important things missing from `webpack-attempt`

File: webpack.config.js [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+};
+
+module.exports = [
+  {
+    ...envConfig,
+    plugins: [
+      new CopyPlugin([
+        { from: "src", to: "", ignore: ["*.js", "*.jsx", "*.ts", "*.tsx"] }
+      ])
+    ],
```
Comment (lusbenjamin) [inline webpack.config.js]: when we used `babel` CLI to build, we passed the `--copy-files` flag to "process" all of our static assets into the `lib` directory. This plugin replaces that part of our build for now. (Eventually all of our static assets should move into the webpack manifest as we move them to local imports, and we may no longer need this.)

File: webpack.config.js (hunk 2 above)
Comment (lusbenjamin) [reply webpack.config.js]: https://webpack.js.org/configuration/node/#node__dirname

Comment (lusbenjamin) [reply webpack.config.js]: https://codeburst.io/use-webpack-with-dirname-correctly-4cad3b265a92

Comment (Manuel-777) [reply webpack.config.js]: Ah, I see why I had issues with absolute paths in the code now!

**Comment by lusbenjamin** [comment]
another missing image due to dynamic reference to static asset: 
image

**Comment by lusbenjamin** [comment]
If we want to merge a better working version of this, we can add the `copy-webpack-plugin` back to our build (it will move all of the assets to `lib`, ensuring the old style of dynamic addressing will work).

**Comment by lusbenjamin** [comment]
now stacked on #754

File: src/window_background/labels.js [hunk 4]
```diff
@@ -9,7 +9,7 @@ import {
 } from "../shared/constants";
 import db from "../shared/database";
 import { playerDb } from "../shared/db/LocalDatabase";
-import CardsList from "../shared/cards-list";
+import CardsList from "../shared/cardsList";
```
Comment (lusbenjamin) [inline src/window_background/labels.js]: 🐛🐛🐛port this to develop immediately?

Comment (lusbenjamin) [reply src/window_background/labels.js]: now #776

**Comment by lusbenjamin** [comment]
closing this for now. We may revisit this post-React conversion when Hot Reloading is a more compelling benefit.

**Comment by Manuel-777** [comment]
> closing this for now. We may revisit this post-React conversion when Hot Reloading is a more compelling benefit.

Agree? Also this had some structural challenges on its implementation since we need to render 2 windows and pack two more.

**Pull request closed without merging**